File: client/src/shared/models/BugUpdateRequestPayload.ts
import { StatusPayload } from './StatusPayload';
import { BugPayload } from './BugPayload';

export default class BugUpdateRequestPayload {
    id: number | null;
    name: string;
    description: string;
    status: StatusPayload | null;

    constructor() {
        this.id = null;
        this.name = '';
        this.description = '';
        this.status = null;
    }

    public static from(bug: BugPayload): BugUpdateRequestPayload {
        const request: BugUpdateRequestPayload = new BugUpdateRequestPayload();
        request.id = bug.id;
        request.name = bug.name;
        request.description = bug.description;
        request.status = bug.status;
        return request;
    }
}

File: api/src/main/java/com/jlopez/bugtracker/model/BugUpdateRequestPayload.java
package com.jlopez.bugtracker.model;

import java.io.Serializable;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class BugUpdateRequestPayload implements Serializable {

  private static final long serialVersionUID = 8143115377640633193L;

  @NotNull
  private final Long id;

  @NotNull
  @Size(max = 255, message = "The title cannot be longer than 255 characters")
  private final String name;

  private final String description;

  @NotNull
  private final StatusPayload status;
}
